src/App.js

```diff
@@ -14,11 +15,99 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  goodsSelector = (goods) => {
+    if (goods.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (goods.length === 1) {
+      return `${goods[0]} is selected`;
+    }
+
+    return `${goods.slice(0, goods.length - 1).join(', ')}`
+      + ` and ${goods[goods.length - 1]} are selected`;
+  }
+
+  removeGood = (good) => {
+    this.setState(state => ({
+      selectedGoods: state.selectedGoods.filter(item => item !== good),
+    }));
+  }
+
+  addGood = (good) => {
+    this.setState(state => ({
+      selectedGoods: [...state.selectedGoods, good],
+    }));
+  }
+
+  clearGoods = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <div className="Title">
+          <h1 className="Title-text">
+            {this.goodsSelector(selectedGoods)}
+          </h1>
+          {selectedGoods.length > 0 && (
+            <button
+              type="button"
+              className="Title-button"
+              onClick={this.clearGoods}
+            >
+              X
+            </button>
+          )
+          }
+        </div>
+
+        <ul className="List">
+          {goodsFromServer.map(good => (
+            <li key={good} className="List-item">
+              <span>{good}</span>
+              {!selectedGoods.includes(good)
+                && (
```

Comment:
```suggestion
              {!selectedGoods.includes(good) && (
```

Reply:
you can create some variable (`isIncluded` for example) inside callback in `map` method before you return `<li>`, and use this variable in all places where you need it. with this approach, you can use the method includes only once